node/src/components/chain_synchronizer.rs

```diff
@@ -288,6 +291,23 @@ where
     ) -> Effects<Self::Event> {
         debug!(?event, "handling event");
         match event {
+            Event::SyncToGenesisResult(result) => {
+                // TODO[RC]: When all fetch operations are unified, rely on the single
+                // `Error::AttemptsExhausted` variant.
+                if matches!(*result, Err(Error::AttemptsExhausted))
+                    | matches!(*result, Err(Error::FetchHeadersBatch(ref err)) if matches!(err, FetchBlockHeadersBatchError::AttemptsExhausted))
+                    | matches!(*result, Err(Error::FetchTrie(err)) if matches!(err, FetchTrieError::AttemptsExhausted))
```

Comment:
You're already within a `match`, so how about something like this instead of a lot of `matches!`?
```rust
Event::SyncToGenesisResult(Err(Error::AttemptsExhausted)) | Event::SyncToGenesisResult(Err(Error::FetchHeadersBatch(FetchBlockHeadersBatchError::AttemptsExhausted))) | ... => { ... }
Event::SyncToGenesisResult(_) => Effects::new(),
```

Reply:
This to me sounds like a good suggestion :+1:

Reply:
That is a bit tricky due to `SyncToGenesisResult` now containing a `Box`. Unfortunately [box patterns](https://doc.rust-lang.org/beta/unstable-book/language-features/box-patterns.html) are unstable unstable. I'd prefer to leave it as is, also because the error structure will eventually be reworked in https://github.com/casper-network/casper-node/issues/3287.

Wdyt?

Reply:
Ok, I'll bors as is, we can discuss again later.